lib/src/presentation/view/movies_by_genres_view.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:flutter/material.dart';
+
+import '../../data/model/genre_model.dart';
+import '../../domain/entity/genre.dart';
+import '../../domain/entity/movie.dart';
+import '../bloc/movie_with_genre_bloc.dart';
+import '../widget/genre_drawer.dart';
+import '../widget/inherited_movie_bloc.dart';
+import '../widget/movies_by_genre_list_view.dart';
+
+class MoviesByGenreView extends StatefulWidget {
+  const MoviesByGenreView({
+    super.key,
+    required this.movieWithGenreBloc,
+  });
+
+  final MovieWithGenreBloc movieWithGenreBloc;
+  static const noGenreTitle = 'Null Genre';
+
+  @override
+  State<MoviesByGenreView> createState() => _MoviesByGenreViewState();
+}
+
+class _MoviesByGenreViewState extends State<MoviesByGenreView> {
+  @override
+  void dispose() {
+    widget.movieWithGenreBloc.dispose();
```

Comment:
I don't think a widget should dispose an object passed as a parameter. It should only dispose its inner objects, thus having control over their lifecycles.

Reply:
where should the dispose be called then? The MovieBlocs are instantiated in the AppRoutes class, but that class is not a widget. The dispose is done at the top of the widget hierarchy, the only alternative i can think of is to make the Views instantiate the blocs instead of getting them as parameters. This is against DI but we never actually need to mock the Blocs so I could just pass the repo (which is whats actually mocked) as a parameter and instantiate the bloc in the actual view widget. I like this option better than the current implementation tbh, instantiating the blocs outside of the views is unwieldy and pretty useless. Do you agree or is there a better option?

Reply:
Yes, I was thinking about the same... I do agree with creating the bloc inside the view itself. With a dependency injection / service locator package you should overcome the DI thing of this approach though (not for this PR).

Or you can create [BlocProviders](https://medium.com/codechai/architect-your-flutter-project-using-bloc-pattern-part-2-d8dd1eca9ba5#:~:text=Now%20its%20time%20to%20implement%20the%20Scoped%20Instance%20BLoC%20approach) and wrap your widgets with them. 
In that article, they dispose the bloc in the child widget, but you could try to dispose it in the BlocProvider itself! That should work, AFAIK.

Either approach would be fine from my side :) 
Also, what do you think @FranciscoGaraventa ?

Reply:
ooh i like the bloc provider approach, thanks for the tip! now I dont need to wrap the listwidgets with an inherited bloc, its already at the top level and both the views and the list widgets can use it without needing parameters. Implemented this approach

Reply:
done, dispose is now called from Provider, not views